apps/ttalps_histogrammer.cpp

```diff
@@ -7,26 +7,33 @@
 #include "TTAlpsObjectsManager.hpp"
 #include "HistogramsFiller.hpp"
 #include "UserExtensionsHelpers.hpp"
+#include "ArgsManager.hpp"
 
 using namespace std;
 
-void CheckArgs(int argc, char **argv) {
-  if (argc != 2 && argc != 4 && argc != 6) {
-    fatal() << "Usage: " << argv[0] << " config_path"<<endl;
-    fatal() << "or"<<endl;
-    fatal() << argv[0] << " config_path input_path output_path"<<endl;
+int main(int argc, char **argv) {
+  
+  auto args = make_unique<ArgsManager>(argc, argv);
+  // check if optional value "config" is present
+  if (!args->GetString("config").has_value()) {
+    fatal() << "No config file provided" << endl;
     exit(1);
   }
-}
 
-int main(int argc, char **argv) {
-  CheckArgs(argc, argv);
-  ConfigManager::Initialize(argv[1]);
+  ConfigManager::Initialize(args->GetString("config").value());
   auto &config = ConfigManager::GetInstance();
-
-  if(argc == 4 || argc == 6){
-    config.SetInputPath(argv[2]);
-    config.SetOutputPath(argv[3]);
+  
+  if (args->GetString("input_path").has_value()) {
+    config.SetInputPath(args->GetString("input_path").value());
+  }
+  if (args->GetString("output_path").has_value()) {
+    config.SetHistogramsOutputPath(args->GetString("output_path").value());
+  }
+  if (args->GetString("output_hists_path").has_value()) {
+    config.SetHistogramsOutputPath(args->GetString("output_hists_path").value());
+  }
+  if (args->GetString("output_trees_path").has_value()) {
+    config.SetTreesOutputPath(args->GetString("output_trees_path").value());
```

Comment:
Following the comment above, we could also move the check of whether the value exists or is null to the config manager.

Reply:
Yeah I agree would be good to clean this up a bit